src/cephadm/cephadm.py

```diff
@@ -3188,6 +3204,13 @@ def get_container(ctx: CephadmContext,
         # to configfs we need to make this a privileged container.
         privileged = True
         set_pids_limit_unlimited(ctx, container_args)
+        # if /etc/hosts doesn't exist, we can be confident
+        # users aren't using it for host name resolution
+        # and adding --no-hosts avoids bugs created in iscsi daemons
+        # by modifications podman makes to /etc/hosts
+        if not os.path.exists('/etc/hosts'):
+            if isinstance(ctx.container_engine, Podman):
```

Comment:
Nit: Since this is podman-related issue, I'd swap these conditions so that the container engine one comes first.  There is no need to check whether `/etc/hosts` if using docker.

Reply:
makes sense